### DISCUSSION TITLE
Docker-Build CI/CD Pipeline

### DISCUSSION THREAD

**Discussion by rajavishah** [bilayer-containers/bilayers#12] [Q&A] [answered]
For our CI/CD setup, we've agreed in building our pipeline with Github Actions, to maintain everything of Github ecosystem.

I believe that we are structuring multiple steps within a single job focused on Docker builds. This job will include running sessions defined in our noxfile and subsequently launching two Docker containers:
     1. Algorithm-Specific Container: Built from instructions in our Dockerfile, under `algorithm-specific` folder.
     2. Interface-Specific Container: Built from the base Dockerfile, which is under `Build` folder.
     
I have one concern in my mind, whether to use checkout@v2 or checkout@v3 for checking out our repository code. Based on the key differences:
- checkout@v3 offers more advanced features like specifying branches (ref), support for sparse checkouts.
So, maybe, I'm more inclined towards using checkout@v3. 

Any suggestions @gnodar01?

**Comment by gnodar01** [comment]
Is there any reason to use checkout@v2. If not, then checkout@v3

**Reply by rajavishah** [reply, marked as answer]
Noted, then am going for checkout@v3
